nox/tasks.py

```diff
@@ -60,14 +61,17 @@ def load_nox_module(global_config: Namespace) -> Union[types.ModuleType, int]:
         # This will ensure that the Noxfile's path is on sys.path, and that
         # import-time path resolutions work the way the Noxfile author would
         # guess.
-        os.chdir(os.path.realpath(os.path.dirname(global_config.noxfile)))
+        os.chdir(noxfile_parent_dir)
         return importlib.machinery.SourceFileLoader(
             "user_nox_module", global_config.noxfile
         ).load_module()
 
     except (VersionCheckFailed, InvalidVersionSpecifier) as error:
         logger.error(str(error))
         return 2
+    except FileNotFoundError:
+        logger.error(f"noxfile.py not found in {noxfile_parent_dir!r}.")
+        return 2
```

Comment:
We need to generalize this error message for the case where the Noxfile is not named `noxfile.py`.

I'd suggest to use the exception handler for `IOError` below this one as a template:

```suggestion
    except FileNotFoundError:
        logger.error("Failed to load Noxfile {}: no such file".format(global_config.noxfile))
        return 2
```

Replacing `logger.exception` with `logger.error` avoids the traceback.

Reply:
Good spot!, I'll tweak that now.